src/ima.rs

```diff
@@ -57,14 +58,16 @@ impl ImaMeasurementList {
 /// was read and the current number of entries in the file.
 pub(crate) fn read_measurement_list(
     ima_ml: &mut ImaMeasurementList,
-    filename: &Path,
+    ima_file: &Option<Arc<Mutex<File>>>,
```

Comment:
Could you explain why `Mutex` is necessary here? Is it possible that the file is accessed from multiple threads?

Reply:
Yeah, I understand that, but in that case I wonder if it might make more sense to handle locking in `quotes_handler.rs` (and if it's not feasible at least have some comment in `ima.rs` why this interface involves locking)?

Reply:
> Yeah, I understand that, but in that case I wonder if it might make more sense to handle locking in `quotes_handler.rs` (and if it's not feasible at least have some comment in `ima.rs` why this interface involves locking)?

I have to take some time and think if there is a better way and if your suggestion is feasible. Right now I can't see another way to open a file before dropping the privileges and later share the handle in multiple threads without using a mutex on the handle. That would be possible if I can clone the handle in each thread and operate on them independently. The content of the file is accessed only for reading, but the handle is modified by seek operations. I need to test this.

Reply:
> Right now I can't see another way to open a file before dropping the privileges and later share the handle in multiple threads without using a mutex on the handle.

I'm not saying that using a mutex is a problem, but suggest calling `.lock()` in `integrity` (in quotes_handler.rs) rather than `read_measurement_list` (in ima.rs). The lock for the other argument (`ima_ml`) is already implemented that [way](https://github.com/keylime/rust-keylime/pull/364/files#diff-e5477179c2c236993dfa1103291d887c44225ca86f91702cba389518fe2eaca4L242).

Here is my [attempt](https://github.com/ueno/rust-keylime/commit/1b0bbabec6423af2f8cc28923c9cba50d6553523) (totally untested).

> That would be possible if I can clone the handle in each thread and operate on them independently.

I am not sure why this mechanism is necessary. Maybe I am missing something but afaics the only code path that could access the file object is `integrity` → `read_measurement_list`, which is even protected by `ima_ml.lock()` so there is only one thread that could take this path.

> The content of the file is accessed only for reading, but the handle is modified by seek operations.

That makes me think that the handle could be generalized as `Read + Seek`, but maybe overkill :-)

Reply:
Ok, I think I understand now what you were asking. I'll try you approach and move the `.lock()`.

Reply:
Modified as suggested. I followed the same approach for the measured boot log file.